crates/electrum/tests/test_electrum.rs

```diff
@@ -0,0 +1,240 @@
+use anyhow::Result;
+use bdk_chain::{
+    bitcoin::{hashes::Hash, Address, Amount, ScriptBuf, WScriptHash},
+    keychain::Balance,
+    local_chain::LocalChain,
+    ConfirmationTimeHeightAnchor, IndexedTxGraph, SpkTxOutIndex,
+};
+use bdk_electrum::{ElectrumExt, ElectrumUpdate};
+use electrsd::bitcoind::bitcoincore_rpc::RpcApi;
+use electrum_client::ElectrumApi;
+use std::time::Duration;
+use testenv::TestEnv;
+
+fn wait_for_block(env: &TestEnv, client: &electrum_client::Client) -> Result<()> {
+    client.block_headers_subscribe()?;
+    let mut delay = Duration::from_millis(64);
+
+    loop {
+        env.electrsd.trigger()?;
+        client.ping()?;
+        if client.block_headers_pop()?.is_some() {
+            return Ok(());
+        }
+
+        if delay.as_millis() < 512 {
+            delay = delay.mul_f32(2.0);
+        }
+        std::thread::sleep(delay);
+    }
+}
```

Comment:
I think this can be a method on `TestEnv`. Maybe rename it to `wait_until_electrum_sees_block` or something more descriptive. It's also immediately apparent why this is useful so please write some docs!